pkg/services/ngalert/writer/prom.go

```diff
@@ -66,9 +66,15 @@ func PointsFromFrames(name string, t time.Time, frames data.Frames, extraLabels
 
 	points := make([]Point, 0, len(col.Refs))
 	for _, ref := range col.Refs {
-		fp, empty, _ := ref.NullableFloat64Value()
-		if empty || fp == nil {
-			return nil, fmt.Errorf("unable to read float64 value")
+		fp, empty, err := ref.NullableFloat64Value()
+		if empty {
+			return nil, fmt.Errorf("empty frame")
+		}
+		if fp == nil {
+			return nil, fmt.Errorf("nil frame")
```

Comment:
This pointer being nil is only true if `empty` or `err` are non-nil (see [here](https://github.com/grafana/dataplane/blob/afa23908e3e3b9d69ea22f00ec56e8c3eb60a21b/sdata/numeric/numeric.go#L44-L56)), so we probably don't need this check.

Reply:
`f` might still be nil if `err` is nil actually ([NullableFloatAt](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/data/field.go#L383))

Reply:
I think that's not quite true, I was able to produce a query that bypassed the other checks and still returned a nil fp.
![2024-09-25-095656_387x291_scrot](https://github.com/user-attachments/assets/15633e91-d76e-4bf3-927e-94c0bfc91a44)

Basically forcing a nil dimension. Without this check, we panic due to a nil deref.

Reply:
We even had a test for this already.

Reply:
Yep looks like I was mistaken, let's definitely leave the nil check in.